alacritty_terminal/src/config/mod.rs

```diff
@@ -418,17 +398,35 @@ impl Default for DefaultTrueBool {
     }
 }
 
+fn defaulted<T, E>(err: E) -> T
+    where T: Default, E: Display
+{
+    error!("Problem with config: {}; using default value", err);
+    T::default()
+}
+
 pub fn failure_default<'a, D, T>(deserializer: D) -> Result<T, D::Error>
 where
     D: Deserializer<'a>,
     T: Deserialize<'a> + Default,
 {
-    let value = serde_yaml::Value::deserialize(deserializer)?;
-    match T::deserialize(value) {
-        Ok(value) => Ok(value),
-        Err(err) => {
-            error!("Problem with config: {}; using default value", err);
-            Ok(T::default())
-        },
-    }
+    Ok(T::deserialize(Value::deserialize(deserializer)?).unwrap_or_else(defaulted))
+}
+
+pub fn or_none<'de, T, D>(deserializer: D) -> Result<Option<T>, D::Error>
+    where D: Deserializer<'de>, T: Deserialize<'de> + Default
+{
+    Ok(match Value::deserialize(deserializer)? {
+        Value::String(ref value) if value.to_lowercase() == "none" => None,
+        value => Some(T::deserialize(value).unwrap_or_else(defaulted))
+    })
+}
+
+pub fn struct_or_str<'de, T, D>(deserializer: D) -> Result<T, D::Error>
+where D: Deserializer<'de>, T: Deserialize<'de> + FromStr<Err = Void> + Default
```

Comment:
There's no need to use `Void` here, especially since it's an external dependency for something extremely trivial. Instead we should just use `Err = ()`, which is more idiomatic Rust anyways.

Reply:
`Void` isn't unit, but an uninhabited type.  This means an error is impossible, which is why it's safe to call `unwrap`.  The `Void` type is just `enum Void{}`, but the equality constraint presumably requires equality of types.  The `void` crate ensures everyone uses the same type.

Reply:
Requiring an external dependency for something so simple is still pretty fragile. In theory this should be done with the never type, though that's not available yet.

However, in this example we could just require `From<&str>` or `From<String>`. This can never throw an error and the unwrap is actually safe since there is none (unlike Void which is vulnerable to refactorings).